payments/schema/mutations.py

```diff
@@ -487,7 +487,7 @@ def mutate_and_get_payload(cls, root, info, **input):
         order_number = input.get("order_number", None)
         try:
             order = Order.objects.get(order_number=order_number)
-            if order.status != OrderStatus.WAITING:
+            if order.status not in OrderStatus.get_waiting_statuses():
```

Comment:
Should we only allow cancelling `OFFERED` orders? Wouldn't make much sense to cancel only drafted ones.